### PR TITLE
JIT: improve stress range experience

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Use `compMethodHash` for stress and inline range inclusions.

Base all stress decisions on the hash of the root method.

Indicate if stress is potentially active for a method in the one-line output
from `DumpJittedMethods`. Indicate if stress is active but this method is
excluded from stress by name or hash. Show the decimal value of the hash.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
cc @dotnet/jit-contrib 

This may cause jitstress to stress some methods a bit differently, so I'll run some of those tests and see if anything new pops up.

**Comment by AndyAyersMS** [comment]
Need to rework a bit to get rid of gotos.

**Comment by AndyAyersMS** [comment]
Kicked off runtime-coreclr jitstress tests.

**Comment by AndyAyersMS** [comment]
All stress runs are failing -- need to fix some initialization ordering.

File: src/coreclr/src/jit/compiler.cpp [hunk 1]
```diff
@@ -5991,11 +6075,11 @@ int Compiler::compCompileHelper(CORINFO_MODULE_HANDLE            classPtr,
...
 
 #ifdef DEBUG
-    if (JitConfig.DumpJittedMethods() == 1 && !compIsForInlining())
+    if ((JitConfig.DumpJittedMethods() == 1) && !compIsForInlining())
     {
-        printf("Compiling %4d %s::%s, IL size = %u, hash=%08x %s\n", Compiler::jitTotalMethodCompiled,
+        printf("Compiling %4d %s::%s, IL size = %u, hash=0x%08x (%u) %s%s\n", Compiler::jitTotalMethodCompiled,
                info.compClassName, info.compMethodName, info.compILCodeSize, info.compMethodHash(),
-               compGetTieringName());
+               info.compMethodHash(), compGetTieringName(), compGetStressMessage());
```
Comment (CarolEidt) [inline src/coreclr/src/jit/compiler.cpp]: I'm curious why you're printing out the method hash in both hex and decimal.

**Review by CarolEidt — APPROVED** [review]
LGTM - just a question

File: src/coreclr/src/jit/compiler.cpp (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/src/jit/compiler.cpp]: No good reason. I should remove the hex.

The range parser parses its list entries as decimal, so the hex value isn't super useful.

Comment (briansull) [reply src/coreclr/src/jit/compiler.cpp]: COMPLUS_JitHashDump and COMPLUS_NgenHashDump both take hex values to select the method to dump.

Comment (AndyAyersMS) [reply src/coreclr/src/jit/compiler.cpp]: Right, all the CONFIG_INTEGERs parse as hex. There are a few others...

If we want to standardize on this (which is fine by me) we should update the range parser to also parse hex.

I find it a less intuitive to do manual range isolation in hex, but probably would get used to it pretty quickly.

Comment (briansull) [reply src/coreclr/src/jit/compiler.cpp]: I prefer seeing the method hashes as 8 digit hex values.

Comment (AndyAyersMS) [reply src/coreclr/src/jit/compiler.cpp]: Let's standardize on hex then... I will have to update the range parser and fix a few other places.

@CarolEidt ok by you?

**Comment by AndyAyersMS** [comment]
Last run's jitstress failures looks comparable to recent results, so this seemingly didn't unearth anything new.

Updated to parse range strings in hex, so one can do:
```
set COMPlus_JitStress=2
set COMPlus_JitStressRange=0-80000000
set COMPlus_DumpJittedMethods=1
...
Compiling    0 System.SpanHelpers::IndexOf, IL size = 899, hash=0x576e6ccc FullOpts JitStress
Compiling    1 System.Text.Unicode.Utf16Utility::GetPointerToFirstInvalidChar, IL size = 1107, hash=0x5e6a3c37 FullOpts JitStress
Compiling    2 System.Runtime.Intrinsics.Vector128::CreateScalarUnsafe, IL size = 21, hash=0xa7141c95 FullOpts NoJitStress(Range)
Compiling    3 System.Runtime.Intrinsics.Vector128::AsInt32, IL size = 7, hash=0x9327a7bb FullOpts NoJitStress(Range)
```
Likewise updated the inline XML to use hex for hashes and tokens.

I've also wanted to be able to trim down inlining and devirt data since we now always jit methods that do lots of inlines -- so changed JitPrintInlinedMethods and JitPrintDevirtualizedMethods into CONFIG_METHODSETs and made corresponding updates.

File: src/coreclr/src/jit/compiler.cpp (hunk 1 above)
Comment (CarolEidt) [reply src/coreclr/src/jit/compiler.cpp]: Fine by me; I don't really care whether it's hex or decimal, but agree that consistency is good :-)

**Pull request merged**